### ISSUE TITLE
Have to run Gyro Up twice when bringing up an S3 bucket with replication configurations

### ISSUE THREAD

**Issue by mlivas** [perfectsense/gyro-aws-provider#195]
When trying to bring up two S3 buckets, with one replicating to the other, I have to run `gyro up` twice due to an error message indicating that "Versioning must be 'Enabled' on the bucket to apply a replication configuration."

Both of my bucket configurations have `enable-versioning: true`.

Here is a simplified version of the configuration I had:

```
aws::s3-bucket example
    name: 'example'
    enable-versioning: true
    @uses-credentials: $primary

    replication-configuration
        role: $(aws::iam-role s3-bucket-replication-role)
        rule
            destination
                bucket: $('aws::s3-bucket' 'example-2')
            end
            id: 'all'
            priority: 1
            status: 'ENABLED'
            delete-marker-replication-status: 'DISABLED'
        end
    end
end

aws::s3-bucket example-2
    name: 'example-2'
    enable-versioning: true
    @uses-credentials: $secondary
end
```

Here is the full error I'm seeing:

```
Error: Can't create aws::s3-bucket example resource!

Caused by: Unexpected error: software.amazon.awssdk.services.s3.model.S3Exception: Versioning must be 'Enabled' on the bucket to apply a replication configuration (Service: S3, Status Code: 400, Request ID: 5AB42835A54C5098)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleErrorResponse(HandleResponseStage.java:115)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleResponse(HandleResponseStage.java:73)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:58)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:41)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:64)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:77)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:39)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.doExecute(RetryableStage.java:113)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.execute(RetryableStage.java:86)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:62)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:57)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26)
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:240)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:96)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:120)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:73)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:44)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:55)
	at software.amazon.awssdk.services.s3.DefaultS3Client.putBucketReplication(DefaultS3Client.java:2883)
	at software.amazon.awssdk.services.s3.S3Client.putBucketReplication(S3Client.java:4837)
	at gyro.aws.s3.BucketResource.saveReplicationConfiguration(BucketResource.java:769)
	at gyro.aws.s3.BucketResource.create(BucketResource.java:473)
	at gyro.core.diff.Create.execute(Create.java:85)
	at gyro.core.diff.Diff.executeChange(Diff.java:407)
	at gyro.core.diff.Diff.executeCreateKeepUpdate(Diff.java:349)
	at gyro.core.diff.Diff.execute(Diff.java:341)
	at gyro.core.command.UpCommand.doExecute(UpCommand.java:55)
	at gyro.core.command.AbstractConfigCommand.doExecute(AbstractConfigCommand.java:135)
	at gyro.core.command.AbstractCommand.execute(AbstractCommand.java:71)
	at gyro.cli.Gyro.run(Gyro.java:195)
	at gyro.cli.Gyro.main(Gyro.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
```

**Comment by mlivas** [comment]
I tried to replicate this again while using the `gyro-0.99.1` release version and had inconsistent results.

My initial attempt had the same error message.

My next attempt, after deleting the previously created S3 buckets and adding `@highlander: true` to my `init.gyro`, I was able to create both S3 buckets with no problems.

Subsequent runs with `@highlander: true` removed from `init.gyro` also were creating the S3 buckets without issue.
